app/models/reporting/data_quality_reports/enrollment.rb

```diff
@@ -0,0 +1,912 @@
+    end
+
+    def calculate_ssn_not_collected ssn_quality:
+      ssn_quality == 99
+    end
+
+    def set_dob_completeness  dob:, dob_quality:, head_of_household:, entry_date:, enrollment_created_date:
+      if calculate_dob_refused(dob_quality: dob_quality)
+        self.dob_refused = true
+        return
+      end
+      if calculate_dob_not_collected(dob_quality: dob_quality)
+        self.dob_not_collected = true
+        return
+      end
+      if calculate_dob_missing(
+        dob: dob,
+        dob_quality: dob_quality,
+        head_of_household: head_of_household,
+        entry_date: entry_date,
+        enrollment_created_date: enrollment_created_date
+      )
+        self.dob_missing = true
+        return
+      end
+      if calculate_dob_partial(dob_quality: dob_quality)
+        self.dob_partial = true
+        return
+      end
+      self.dob_complete = true
+    end
+
+    def calculate_dob_complete dob:, dob_quality:
+      dob_quality.in?(COMPLETE) && dob.present?
+    end
+
+    def calculate_dob_refused dob_quality:
+      dob_quality.in?(REFUSED)
+    end
+
+    def calculate_dob_missing dob:, dob_quality:, head_of_household:, entry_date:, enrollment_created_date:
+      missing = false
+      missing = true if dob.blank?
+      missing = true if dob_quality == 2
+      missing = true if dob.present? && dob < '1915-01-01'.to_date
+      missing = true if dob.present? && dob > enrollment_created_date
+      missing = true if (is_adult?(date: Date.today) || head_of_household) && dob.present? && dob >= entry_date
+
+      return missing
+    end
+
+    def calculate_dob_partial dob_quality:
+      dob_quality == 2
+    end
+
+    def calculate_dob_not_collected dob_quality:
+      dob_quality == 99
+    end
+
+    def set_gender_completeness  gender:
+      if calculate_gender_refused(gender: gender)
+        self.gender_refused = true
+        return
+      end
+      if calculate_gender_not_collected(gender: gender)
+        self.gender_not_collected = true
+        return
+      end
+      if calculate_gender_missing(gender: gender)
+        self.gender_missing = true
+        return
+      end
+      self.gender_complete = true
+    end
+
+    def calculate_gender_refused gender:
+      gender.in?(REFUSED)
+    end
+
+    def calculate_gender_missing gender:
+      gender.blank?
+    end
+
+    def calculate_gender_not_collected gender:
+      gender == 99
+    end
+
+    def set_veteran_completeness  veteran:
+      if calculate_veteran_refused(veteran: veteran)
+        self.veteran_refused = true
+        return
+      end
+      if calculate_veteran_not_collected(veteran: veteran)
+        self.veteran_not_collected = true
+        return
+      end
+      if calculate_veteran_missing(veteran: veteran)
+        self.veteran_missing = true
+        return
+      end
+      self.veteran_complete = true
+    end
+
+    def calculate_veteran_refused veteran:
+      return false unless is_adult?(date: Date.today)
```

Comment:
Actually I think we need to calculate this based on the entry date.  If the client wasn't an adult at entry, they wouldn't have asked if they were a vet.  There's a chance this would get updated later, but I think for data quality we don't want to assume that happened.  Good catch.